### DISCUSSION TITLE
Microlearning

### DISCUSSION THREAD

**Discussion by abotzki** [LiaScript/LiaScript#158] [Q&A] [answered]
Hi all, 

I recently discovered a company 7taps which is promoting microlearning. I think that the concept deserves its place in the educational landscape. 
Here is a link to an example from a production company with workers at assembly lines. It has been shared during a conference. 

https://app.7taps.com/qdLphlRyhk

I think that this concept could be 'reproduced' with Liascript. 

My question is twofold - I saw that you have the option to load the 'app' on the phone. At least, it behaves like an app once you load the course on the phone. Is there an official app or how it is working?

And second, do you see an option to make a layouting with Liascript which is also so slick?  ;-)

**Comment by cagix** [comment]
That's quite interesting. It reminds me in a way of [Anki](https://apps.ankiweb.net/) cards. Only in this case it's for knowledge transfer and not for knowledge assessment.

**Comment by abotzki** [comment]
Thanks for your comments @cagix I used the Anki cards for learning a language lately.

**Reply by andre-dietrich** [reply, marked as answer]
@abotzki thanks for sharing ... I played around with OpenBadges by [ICOBC](https://openbadges.icobc.net) that can be used also to for certificates for courses and micro-credentials ... anyways ...

The LiaScript-Website is a PWA, which stands for Progressive Web App:

https://en.wikipedia.org/wiki/Progressive_web_app

This is basically a WebSite with a manifest file and "service-worker", the service-worker is a JavaScript file and stores all required files within the browser-cache and does a bit more ... But the idea is, that such websites can be installed like native apps, but do not require something like a PlayStore ... Works also on desktop with Chrome ...

Basically it is also possible to export a LiaScript-course into an APK-file that can be installed on Android as a native app, but this requires some configuration with Android-Studio. @SebastianZug demoed this here at minute 48

https://video.tu-freiberg.de/video/E2DTeach40Talk235-Open-Educational-Ressources-28OER29-Vision2C-Vorteile-26amp3B-praktische-Umsetzung-mit-LiaScript-und-OPAL/0e36a414baa3158a6339be43b2e4dbc5

Concerning the slick animations ... I guess this should be already possible, I will have to play a bit with the styles, but you can basically restyle the entire course. CSS can be imported via `link:` and simple changes can be made with the `@style` macro:

``` markdown
<!--
@style
h1 {
    color: red
}
@end
-->

# Hello World
```

Unfortunately I am not a CSS guy, I whish I would, but styling always takes time for me :see_no_evil: ... I will keep this on my todo-list ...

**Reply by abotzki** [reply]
Really nice, I will check the presentation by Sebastian in more detail. 
As to the CSS, I might need to get in touch with some colleagues who are more techy than I. ;-)
I have played a little bit with this to get a more VIB layout: see one example for a course overview page [here](https://liascript.github.io/course/?https://raw.githubusercontent.com/vibbits/functional_analysis_training/master/README.md).  

Note: the fa icons do not always load fine, need to investigate further.

**Comment by andre-dietrich** [comment]
@abotzki concerning the animation ... a starting point could be this (if I got you correct)

``` markdown
<!--
@style

@keyframes zoomIn {
    from {
        transform: scale(0);
        opacity: 0;
        border: 5px solid black;
    }
    to {
        transform: scale(1);
        border: 1px solid black;
        opacity: 1;
    }
}

/* Apply the animation to the main element */
main {
    animation: zoomIn 0.75s ease-in-out;
}

@end
-->

# Course Main Title
```

[LiveEditor-Example](https://liascript.github.io/LiveEditor/?/show/code/H4sIAAAAAAAAA71XW3PbxhV+x69Y0w8RYQG8OG5dSr7IkpLSsSPXUqbuaDT2EliSawFYBAuIoh1lHHvSyyS9zLSPnelrHxUnqt340r8A/qN+ZxckRbv1pO1MQQ65u+d+9tywesbzHOeqzseRwP++GPczHgvNHigVdxP20GF4+pmKqyU9ecYT3VdZ3GE64JFYatZXZkCV8kDm4w5rzs96KgtF1mEX0kOmVSRD1ot4sG8Rjsxvrt4loFV/m1nrXzFb0KA1FXDkOA2XraVpNGb5UDCeyJjnUiUklidMRCIWSc7chhNzObV6htWZeqPp//iCZoJr4cnEU0W+QqyviiR0PO+y45xl66rItGA3icuOzMmpO0OpGb5jgJjrBgbDdZlMZC55JB9YTXRe9HzHuRURe6aFMJrubqhA7y0N8zzVnUYjklwHmUxzfyDzIQikaliGjStTnIyPKmihRRaoJIdpfqBiot421I0QXBsx17nIGrc31zZubvpxWHfgjb5MQgbL2GjIc1I7VVrLXiSgL9u9MeXwTpXqsKPbJ4PZiMOtYAtVwEDntB2KKCVuxiFrOUKru7G5zFJreT5OBdwEtvARKMkTPIoYP+Ay4qSIHqosD4pcQ8zZs/B1th+qUeI4v4DAANdJwmBvDK/uTqFzhSs1ySE85LGXwhEWxxuKTDRGcl82pmTe+lDwXA+FyOtMj5OcH5JSQYZTYS2w/l9mugiGjOuO47R8dkPqXDttn5lgFXBp5lBQuaxIqiO7VQmSzfdhivMZ+6ngALEWILNNmzaAdbzpwxY3gHVzERsqojObtt2AazfmA0FandmVZjn3RJFGioc+GRyLUHJfZQNjfkq7hnWhbrQb7Wbjw0j1eHR3KxXJ3c2wCEzQ4uC20LA/EPruDTVQvj4Y1B26lrNs8xCBp4GlHZuanvfwYfPoCAXnGuJrXF0Wj7RCNARREQqmEPMZ68O5RQa/VC5lOlX7AskqDnO/Yma4tQy3boI4z5GuY7ZvorfPeBFKiuVIdKb4zKA7V3YNbO6EnhzAgiTs8WTfBMUnt25srW004vR8o9X80UUfCzJpJrRthG4eQKEDGQqlcbcsGPJkYFM2wqqAoykEEdO5iMb+KSXa3nmocebKriGe6zEajXz4BFVAGDVGPA+GVw4u9brr/Y9ujz/aGW3O1DB6nGe3C6QmXPiBiFEkjVrlX8qTyaPyafm6fFUeT35VHpcvJl9j/ZpNvsDhs/I7HL1i5VPWbjbfZ+W3OPxu8oSVfyr/Zohely9Z+cfJo8mT8hvDiw6WiAi4L3x2XQ0T9mEGNbM6K5+z8g+Q82hK6pR/hXAcTH5ZnpQvfVb+2QC+LZ+XJ0CH+BMQTX5fPpt8VX5fHrPyGyxeEKN/GP2OgfRy8sXksTHiKQmefEWUz2DB5MnkN1DKQLH+HXbPJ19a06AdBP8a/F8B+++GnwPNjsH0JYgeQ9wJw99J+b3hcExemPwWZHTyuDIC0sARxM+B9DUojydf2vy090e3t0M1qnagZCBqsxrFIuS8Cb6FajUNB6pWJi+20WtlMnCcVfReFkRca3aJ1Wy3QaXIVBR1kxozPXkOQSJ5oYg4+lpbryC4IoUmmBYZYqzGqP3sIPhGQwUiShSvF6lgn4plEYWMp6ngGVVdS2HpkTMhG6EeQvuR6vVILyb6fREgz34+lMg+dACOctzjIcqxOOQU0rNabap6bjkEvDB9jHz1+eeoa6bX2tZHBXDFqzOc/xujZdKnfijQ3YuEvDqzvjIULFasmVTz14c8Q311PoECa9vr3a63lpk+E6L7MRSvAaYYPc/9dz43iyiXMF0Y7Jb/ExRRPBtbO9sz+s/eonJdt4KO34K69nearIBnpz/uG//2U2HzN7i581WFcbiI4Z5eVygSKNdmH/fUemFXYetTDE9zW+RJTnAJ4W2UKak7w/Za7Jz3A54Z7+Yir0OPHyJwFp+WufufFfLBA6FtMq2xHcT6pzhyELCYWELMg1SFXbcv9RAxq/kYWYFijcMhgpWbGSxBB0e0jZCrrnvFxsnuLsaBvT3L2AQFJcr6kNLcca4XyG4ehtSPYuo1qZJJrmlrBh1I60z53NnbY1tJNWy67h2oQTOGNvtAZRkSzNL7FQUDxWac5mM7EPAMumUqGfinOJoqZCoI0vS/02yJ1S0fu7uD3Sry8Q1dKeejSI1E6L9B5pi+LjB8mdq2rkLIXzNZZ0rHtD5gdmehoOkBo3xOoJwuh5S7jtJoZ0gUzF7Gs7EZA3qmmoTLdh4w9JYphN67d+8+qOyY6TQarDsdnans3kDmUv9FAUB5sx7GnIbTzFYmOupuWJSWk4iRLR+o1j7RLr131oLeWzaDP8q2iHSH7baWWXuZnV9m7+8t4xxDhhR0vttqNpdZq00/F/GDRoqoOcIryv+mW3tRt2s8m6rW/sGqXTCAi4BVKsF3zqr13OWrMkmLfLVRbdGBQnlgi/GlWpB7VlEsBioKReJphClKeo3JEHDjo9rl1QaILv/npO0ZqQ3iW5m6DyQ9H9tjvj8bqTELinmcUZsKRZ+cmVZkHRMW7L6tEd7mAY9sVPn3tYOpizoh2kfIc+73Zabzuwnea9k5VsPnnD2H9vYYNyf7S+YMfQti6sbdxIIIkA/2uLaCl0ncgHgT/PHWzmkUcrrVDg0Rz7kNYk07hwhrc4VqrIP9No/jcY1usjbTqcYsCL7bt6BKgDG4Qy/JhXCOFi4YWIhAaedhCq3r21sfG1u1oAhEgzQTqsJLNTvgmTTeJTgFF1BoCa/ZQDFv3ZalgHsNqRVksxn5qm2LryQAlxCX7lX0zfo9cDgVb1RXVYaS8cHCW+b0VUDQTB0qFisqf5Qen9paj8v+v7wBO/8E5nwnfA8RAAA=)

You could define different animations for different elements... `<main>` contains the entire slide ... I simply asked ChatGPT: "make a css zoom in animation"
